src/common/components/debounced-search.tsx

```diff
@@ -23,14 +23,33 @@ export function DebouncedSearch( {
 	inputAriaLabel,
 }: Props ) {
 	const [ searchTerm, setSearchTerm ] = useState( '' );
+	const [ lastEmittedSearchTerm, setLastEmittedSearchTerm ] = useState( '' );
+
+	const callCallback = useCallback(
+		( searchTerm: string ) => {
+			setLastEmittedSearchTerm( searchTerm );
+			callback( searchTerm );
+		},
+		[ callback ]
+	);
+
+	const conditionallyCallCallback = useCallback(
+		( searchTerm: string ) => {
+			if ( searchTerm.length >= debounceCharacterMinimum && searchTerm !== lastEmittedSearchTerm ) {
+				callCallback( searchTerm );
+			}
+		},
+		[ callCallback, debounceCharacterMinimum, lastEmittedSearchTerm ]
+	);
+
 	const debouncedCallback = useMemo(
-		() => debounce( callback, debounceMs ),
-		[ callback, debounceMs ]
+		() => debounce( conditionallyCallCallback, debounceMs ),
+		[ conditionallyCallCallback, debounceMs ]
 	);
 
 	const searchImmediately = () => {
 		debouncedCallback.cancel();
-		callback( searchTerm );
+		callCallback( searchTerm );
```

Comment:
**suggestion (non-blocking):** What do you think about trimming the search term so it will remove any leading or trailing whitespace first? 

When a user enters only space and presses enter, we should show the placeholder and treat it as an empty search term instead of showing the results for the previous search term. 

**Steps to reproduce:** 

1. Search for the term "test".
2. Wait for the search results. 
3. Clear the search input.
4. Press enter and see the placeholder displayed as expected.
5. Enter a single space in the search input and press enter.
6. Notice that the search results for the previous term "test" are displayed instead of treating it as an empty search term.

Reply:
Wow, that's a great catch, Karen! 👏

Reply:
Fantastic idea!! Got it implemented 👍 

There are comments to the effect, but the short of the trimming behavior is...
- We only ever emit trimmed searches
- On enter/search button, we change the input display to show the trimmed term
- On blur, we change the input display to show the trimmed term
- Notably, we don't change the actual input display to the trimmed term on debouncing only. This felt kind of jarring -- the cursor would just jump all over the place! 😆